### PR TITLE
[fix](mtmv)Fix tablet not found when restart fe

### LINKED ISSUE #20094: [Bug] restart fe,mv tablet miss
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

restart fe

select * from mv;

ERROR 1105 (HY000): errCode = 2, detailMessage = (127.0.0.1)[INTERNAL_ERROR]failed to get tablet: 15883, reason: tablet does not exist. 127.0.0.1

### What You Expected?

normal

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zddr**
# Proposed changes

Issue Number: close #20094 

## Problem summary

Describe your changes.

The replayCreateTable restriction must be olapTable. If mv is used, nothing will be done, resulting in no call to invertedIndex.addReplica

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/DatabaseIf.java [hunk 1]
```diff
@@ -172,6 +172,19 @@ default T getTableOrMetaException(String tableName, List<TableIf.TableType> tabl
...
+        for (TableIf.TableType tableType : tableTypes) {
+            if (table.getType() == tableType) {
+                return table;
+            }
+        }
+        throw new MetaNotFoundException(
+                "Tye type of " + tableName + " doesn't match, expected data tables=" + tableTypes);
+
+    }
+
```
Comment (adonis0147) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/DatabaseIf.java:187]: The definition of this method looks similar as the above one (`default T getTableOrMetaException(String tableName, List<TableIf.TableType> tableTypes)
            throws MetaNotFoundException`). We should reuse the same logic.

**Review by adonis0147 — CHANGES_REQUESTED** [review]

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35 seconds
 stream load tsv:          443 seconds loaded 74807831229 Bytes, about 161 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          79.3 seconds inserted 10000000 Rows, about 126K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230526093518_clickbench_pr_151157.html

**Review by adonis0147 — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.17 seconds
 stream load tsv:          444 seconds loaded 74807831229 Bytes, about 160 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          76.1 seconds inserted 10000000 Rows, about 131K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230526162320_clickbench_pr_151503.html

**Pull request merged**